### ISSUE TITLE
Add logging to jira plugin

### ISSUE THREAD

**Issue by jiahaluo** [abcxyz/jvs-plugin-jira#48]
### TL;DR

Context: https://github.com/abcxyz/jvs-plugin-jira/pull/45#pullrequestreview-1603146101

We discovered an issue where logs created using the `pkg/logging` library were not being reflected in cloud logging ([test PR](https://github.com/abcxyz/jvs-plugin-jira/pull/47) and [screencast](https://screencast.googleplex.com/cast/NTUxNzM5MzU2OTc3NTYxNnwxYWQ5ODIzZC1mZg)).
Further investigation is needed to resolve this issue and restore the logger.

### Detailed design

Offline discussed with @yolocs, we'll go with this workaround:

In jvs-jira-plugin, we don't log those errors. Instead, we return the full message as part of the error. In the main program, we do the logging based on the status code. In JVS, if we see an internal error, we log it and return an error without details. Otherwise, we return the raw error message.
